_inc/lib/tonesque.php

```diff
@@ -37,27 +37,7 @@ function __construct( $image_url ) {
 	}
 
 	public static function imagecreatefromurl( $image_url ) {
-	 	// Grab the extension
-		$file = strtolower( pathinfo( $image_url, PATHINFO_EXTENSION ) );
-		$file = explode( '?', $file );
-		$file = $file[ 0 ];
-
-		switch ( $file ) {
-			case 'gif' :
-				$image_obj = imagecreatefromgif( $image_url );
-				break;
-			case 'png' :
-				$image_obj = imagecreatefrompng( $image_url );
-				break;
-			case 'jpg' :
-			case 'jpeg' :
-				$image_obj = imagecreatefromjpeg( $image_url );
-				break;
-			default:
-				return false;
-		}
-
-		return $image_obj;
+		return imagecreatefromstring( file_get_contents( $image_url ) );
```

Comment:
I didn't want to introduce an external dependency into the library, and using `file_get_contents()` was exactly what PHP was internally doing anyway.
Right now someone could be using Tonesque + `Jetpack_Color` stand-alone without issue, as far as I could see.

We could use `wp_remote_get()` if the `$image_url` is prefixed by `http://` or `https://` however would need to use `file_get_contents()` for all other protocols and image paths anyway (ie. `/home/.../image.jpg` or `custom-file-wrapper://path/image.jpg`)